src/Libraries/CoreNodes/List.cs

```diff
@@ -688,6 +689,78 @@ public static IList Transpose(IList lists)
             return transposedList;
         }
 
+        /// <summary>
+        /// Checks if a given list is empty or is a single dimensional list that contains only null values
+        /// </summary>
+        /// <param name="list"></param>
+        /// <returns>Returns true if the input list is empty or contains only null values</returns>
+        public static bool IsNullOrEmpty(IList list)
```

Comment:
The `IsNullOrEmpty` node will return `true` only for a single dimensional array with nulls. So if the array is something like `{null, {null}, null}`, it will return `false`.